src/wallet/test/util.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <wallet/test/util.h>
+
+#include <chain.h>
+#include <key.h>
+#include <test/util/setup_common.h>
+#include <wallet/wallet.h>
+#include <wallet/walletdb.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <memory>
+
+std::unique_ptr<CWallet> CreateSyncedWallet(interfaces::Chain& chain, CChain& cchain, const CKey& key)
```

Comment:
> Fixtures are overused in general right now, causing tests to do unnecessary work, and they make setup code that could be reused in other contexts (benchmarks & fuzz tests, future test & simulation code) harder to understand and less reusable. 

In this specific case the fixture doesn't cause unnecessary work.
`CreateSyncedWallet` function is used in only two places now: 1) in `ListCoinsTestingSetup` and 2) in the new test that uses `TestChain100Setup` fixture anyway.

In general I agree that fixtures are harder to reuse in other contexts but without having a real use case I'm not sure that `CreateSyncedWallet` will actually meet the requirements.

> The current hierarchy of test fixtures that inherit from each other is also confusing and inflexible.

But you are using fixtures anyway right now. So why then split some part of the setup in fixture and some part in a standalone method. Using two separate mechanism to do the test setup is even more confusing. 

Also there is value in maintaining uniformity in how the test setup is done. 
Fixtures are already used extensively in the code base. There are 72 test suites that use fixtures vs 24 that don't. According to my spot check the ones that don't use fixtures also don't use standalone setup function as a way to reuse setup code between tests. 

> The suggestion here also seems a little vague, because I'm not sure what form or members you'd want a generic fixture to have. ListCoinsTestingSetup has more functionality than the new test requires, and less functionality than new tests may require. Would the new fixture only support tests with one synced wallet, or with multiple wallets? Would it require all the wallets to be created before each test starts, or could the tests create the wallets as needed?

My suggestion is to not introduce a second mechanism to do test setups and just reuse existing fixture (probably renaming it to `SyncedWalletTestingSetup`) because it has exactly the right setup. 

The new `SubtractFee` test has the setup exactly equivalent to the setup of `ListCoinsTestingSetup`. `ListCoinsTestingSetup` has also `AddTx` method which is used exactly one time and could easily be inlined or extracted.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/22155#discussion_r673741284

> My suggestion is to not introduce a second mechanism to do test setups and just reuse existing fixture (probably renaming it to `SyncedWalletTestingSetup`) because it has exactly the right setup.

Yes, I see what you're saying here. If you can use fixtures to help with test setup, and you can also use normal functions to also help with test setup, then setup code might be less discoverable, and someone who is trying to write a new test and only looking at the fixtures might miss a function that could be useful. I would not deny that this is a drawback. I just think there are some advantages on the other side. (Functions that don't maintain state can be easier to understand than fixtures that are stateful. Functions can be more flexible than fixtures and in this case let you create many synced wallets instead of one. Functions can be shared across frameworks including boost framework, qt test framework, fuzzing framework, benchmark framework, and other future frameworks, instead of limited to the boost framework. And this change adds a place for more functions to go and more fixture code to be detached even if is not moving right away)